binding/rust/src/rhino.rs

```diff
@@ -0,0 +1,543 @@
+/*
+    Copyright 2021 Picovoice Inc.
+
+    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+    file accompanying this source.
+
+    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations under the License.
+*/
+
+use libc::{c_char, c_float};
+use libloading::{Library, Symbol};
+use std::cmp::PartialEq;
+use std::collections::HashMap;
+use std::convert::AsRef;
+use std::ffi::CStr;
+use std::path::{Path, PathBuf};
+use std::ptr::{addr_of, addr_of_mut};
+use std::sync::Arc;
+
+#[cfg(target_family = "unix")]
+use libloading::os::unix::Symbol as RawSymbol;
+
+#[cfg(target_family = "windows")]
+use libloading::os::windows::Symbol as RawSymbol;
+
+use crate::util::*;
+
+#[derive(Clone, PartialEq, Debug)]
+pub enum BuiltinContexts {
```

Comment:
Rhino doesn't include built-in contexts. Though we do have contexts in the Rhino repo, those are for the demos. The actual packages don't ship context files with them.